src/settings/settingsbase.h

```diff
@@ -0,0 +1,124 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef SETTINGSBASE_H
+#define SETTINGSBASE_H
+
+#include <QSettings>
+
+#include "loghandler.h"
+#include "setting.h"
+#include "settinggroup.h"
+
+/**
+ * @brief The SettingsBase is a singleton class that manages the underlying
+ * QtSetting storage object and serves the as the holder of all settings in the
+ * codebase dynamic or static. The lifetime of this singleton is tied to the
+ * lifetime of the QApplication.
+ *
+ * There are two types of settings currently in this codebase.
+ *
+ * * Static settings are settings defined at build time in a declarative manner
+ * e.g. the SETTING macro or the EXPERIMENTAL_FEATURE macro.
+ * * Dynamic settings are settings which are created on demand at runtime.
+ *
+ * All settings are created using the SettingsFactory, which creates a Setting
+ * object and registers it in the SettingsBase class.
+ *
+ * The only public constructor for Setting objects is the SettingsFactory
+ * constructor, therefore everytime a Setting object is created it is guaranteed
+ * to be registered in the settings base.
+ *
+ */
+class SettingsBase final : public QObject, public LogSerializer {
```

Comment:
This name seems to indicate that it is a base class. Consider renaming it to something like SettingsManager